src/Symfony/Component/Security/Core/User/CurrentUserProvider.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Symfony\Component\Security\Core\User;
+
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Current User Provider.
+ *
+ * This provider gives you the current logged in user.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class CurrentUserProvider
```

Comment:
In my opinion you should not bring that state to your services. You'll still need to check if the user is there and act on it. I think it's a far better solution to just inject the user if you need it in the method that you call. If you really need the user in a deep level of your services, then maybe you shouldn't be using the UserInterface for that.

```
public function someAction(UserInterface $user)
{
    $this->someService($user);
}
```

Right now you're adding a class that does nothing more than removing some boiler plating code. It also makes services harder to use on the commandline. The current name `CurrentUserProvider` also implies that the service has a state, which it shouldn't have.

I have a very big :-1: for this class as there are already other _(better)_ solutions available.